enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+
+### Goals
+
+1. Design meaningful metrics to collect user experience and behavior with the installer
+
+### Non-Goals
+
+1. Collected metrics will not allow analysis of individual installer invocations and will not display detailed data such as log or error messages. We won't be able to answer "how long did an individual installation take?" but rather 95% of installs finish within, say, 40 minutes.
+2. Installer instrumentation/internal implementation will be discussed in a separate design/enhancement
+3. Pushgateway implementation will be discussed in a separate design/enhancement
+
+## Proposal
+
+### User Stories
+
+#### Telemetry Analyst
+Some examples:
+
+As a telemetry analyst I would like to create queries which analyze aggregate installation data to determine how long the installation process takes in total and broken down by different stages so that I can better understand user experience, expose any abnormalities, and compare different versions of the installer.
+
+As a telemetry analyst I would like to create queries which show aggregate counts of failure by stage of installation so that we can better understand where users are experiencing problems. (This will not provide  specific error messages/logs).
+
+As a telemetry analyst I would like to see how often users modify ignition configs or manifests so that I can better understand how users are interacting with the installer. [(see open question #3)](open-questions)
+
+As a telemetry analyst I would like to see how often the installer is being run on certain operating systems so I can better understand our user environments and needs.
+
+### Implementation Details/Notes/Constraints
+The installer does not fit a typical Prometheus use case in that:
+* metrics will be pushed rather than scraped
+* Prometheus labels will be (ab)used to collect information 
+* the aggregation push gateway will be used to form the metrics into a useful time series
+
+The proposed metrics fall into three categories:
+
+#### Commands
+
+##### cluster_installation_create 
+`cluster_installation_create` represents an invocation of `openshift-install create cluster`. The values for the labels would be collected throughout the execution of the command and a single sample would be pushed per installer run:
+
+```
+# HELP cluster_installation_create represents a single run of create cluster by the OpenShift installer.
+# TYPE cluster_installation_create histogram
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="15"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="20"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="25"} 0
+cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="30"} 1
```

Comment:
```suggestion
cluster_installation_create_bucket{platform="aws",result="success",version="4.2",os="linux",le="30"} 0
```
example below says between 30-35 mins

Reply:
Hm, isn't this largely correct?

Technically, I think this depends on the [rounding in the installer](https://golang.org/pkg/time/#Duration.Round). So if an install took 29 minutes and 30 seconds, the value would be rounded to 30 and `le="30"` (less than or equal to 30) would be true. So really the window should be 29:30-34:29 seconds, inclusive. 

Do correct me if I'm wrong.

Reply:
Ah my bad. Didn't see the sum part.